include/deal.II/fe/fe_values.h

```diff
@@ -4285,6 +4341,23 @@ namespace FEValuesViews
         return return_value;
       }
   }
+
+  template <int dim, int spacedim>
+  inline
+  typename Tensor<2, dim, spacedim>::gradient_type
+  Tensor<2, dim, spacedim>::gradient(const unsigned int shape_function,
+                                     const unsigned int q_point) const
+  {
+    Assert (shape_function < fe_values->fe->dofs_per_cell,
+            ExcIndexRange (shape_function, 0, fe_values->fe->dofs_per_cell));
+    Assert (fe_values->update_flags & update_gradients,
+            (typename FEValuesBase<dim,spacedim>::ExcAccessToUninitializedField("update_gradients")));
+
+    const int snc = shape_function_data[shape_function].single_nonzero_component;
+    AssertThrow(false, ExcNotImplemented());
```

Comment:
Remove the `snc` variable since it's not used and will only result in a warning.

Reply:
> since it's not used

it's not used because the function is not implemented 😄

Reply:
I know, but the compiler doesn't :-)